pyke/lightcurve.py

```diff
@@ -182,20 +189,38 @@ def get_lightcurve(self, flux_type, centroid_type='MOM_CENTR'):
                                     campaign=self.campaign,
                                     quarter=self.quarter,
                                     mission=self.mission,
-                                    cadenceno=self.cadenceno)
+                                    cadenceno=self.cadenceno,
+                                    id = self.hdu[0].header['KEPLERID'])
         else:
             raise KeyError("{} is not a valid flux type. Available types are: {}".
                            format(flux_type, self._flux_types))
 
-    def _quality_mask(self, quality_bitmask):
+
+    def _quality_mask(self, bitmask, mask):
```

Comment:
This function can just take `bitmask` as a parameter.  If `bitmask` is an integer then the function will return `data['SAP_QUALITY'] & bitmask`, if `bitmask` is a string then it will use `data['SAP_QUALITY'] & bitmask_dict[bitmask]`.

Reply:
👍